src/main/java/Bob/util/controller/Command.java

```diff
@@ -0,0 +1,314 @@
+        String formatId = String.format("%-4s", (id + 1) + ".");
+        UI.printTab(formatId);
+    }
+
+    /**
+     * Displays the current tasks and their statuses.
+     *
+     * @param list a Task class list.
+     */
+    public static void displayList(ArrayList<Task> list) {
+        UI.printBorder();
+        int count = list.size();
+
+        if (count == 0) {
+            UI.printError(UI.MESSAGE_NO_TASKS);
+            return;
+        }
+        UI.printlnTab(UI.MESSAGE_TASK_LIST);
+        for (int i = 0; i < count; i++) {
+            printListId(i);
+            System.out.println(list.get(i));
+        }
+        UI.printBorder();
+    }
+
+    /**
+     * Updates the completion status of an indicated task.
+     *
+     * @param list    a Task class list.
+     * @param command the command containing the task id to be updated.
+     * @param isDone  the status to be updated to.
+     */
+    public static void updateStatus(ArrayList<Task> list, String command, boolean isDone) {
+        UI.printBorder();
+        String[] commandToken = Parser.parseCommand(command, DELIMIT_COMMAND);
+
+        if (commandToken[1] == null) {
+            UI.printError(UI.MESSAGE_INVALID_ARGC);
+            return;
+        }
+        try {
+            int id = Parser.stringToInt(commandToken[1]);
+            isValidCurrentId(list, id);
+            Task target = list.get(id - 1);
+            target.setDone(isDone);
+            LoadSave.saveData(list);
+
+            if (isDone) {
+                UI.printlnTab(UI.MESSAGE_TASK_MARKED);
+            } else {
+                UI.printlnTab(UI.MESSAGE_TASK_UNMARKED);
+            }
+            System.out.println("\t" + target);
+            UI.printBorder();
+        } catch (NumberFormatException e) {
+            UI.printError(UI.MESSAGE_INVALID_ID_FORMAT);
+        } catch (BobInvalidIdException e) {
+            UI.printError(UI.MESSAGE_INVALID_ID_NUMBER);
+        }
+    }
+
+    public static void findTasks(ArrayList<Task> list, String command) {
+        UI.printBorder();
+        String[] commandToken = Parser.parseCommand(command, DELIMIT_COMMAND);
+
+        if (commandToken[1] == null) {
+            UI.printError(UI.MESSAGE_INVALID_ARGC);
+            return;
+        }
+        int count = list.size();
+        if (count == 0) {
+            UI.printError(UI.MESSAGE_NO_TASKS);
+            return;
+        }
+        boolean taskFound = false;
+        for (int i = 0; i < count; i++) {
+            Task currentTask = list.get(i);
+            if (currentTask.getDescription().contains(commandToken[1])) {
+                if (!taskFound) {
+                    UI.printlnTab(UI.MESSAGE_FOUND_MATCHING_TASK);
+                    taskFound = true;
+                }
+                printListId(i);
+                System.out.println(currentTask);
+            }
+        }
+        if (!taskFound) {
+            UI.printlnTab(UI.MESSAGE_FOUND_NO_MATCH);
+        }
```

Comment:
It seems you have a number of conditions to check and [arrowhead code](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting) is emerging from lines 123-130. Try applying [SLAP](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) to improve code readability.